File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sl = len(s)
        tl = len(t)
        if sl != tl:
            return False
        
        sd = {}
        td = {}
        
        for c in s:
            if c not in sd:
                sd[c] = 0
            sd[c] += 1
            
        for c in t:
            if c not in td:
                td[c] = 0
            td[c] += 1
            
        return sd == td

File: 0242-valid-anagram/0242-valid-anagram.go
func isAnagram(s string, t string) bool {
    ls := len(s)
    lt := len(t)
    if ls != lt {
        return false
    }
    
    ds := map[rune]int{}
    dt := map[rune]int{}
    
    for _, ch := range s {
        ds[ch]++
    }
    for _, ch := range t {
        dt[ch]++
    }
    
    for k, v := range ds {
        if dt[k] != v {
            return false
        }
    }
    
    return true
}